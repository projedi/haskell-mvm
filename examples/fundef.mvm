double k(int i, double j);

void f(int i, double j, string str) {
  i = i + 1;
  j = j * 1.4;
  str = "def";
  print(i, " ", j, " ", str, "\n");
}

f(0, 3.1, "abc");
int i;
double j;
string str;
i = 0;
j = 3.1;
str = "abc";
f(i, j, str);
print(i, " ", j, " ", str, "\n");

double g(int i, double j) {
  return i * j;
}

print(g(2, 1.3), "\n");

void h(int i) {
  if (i == 0)
    return;
  print(i, "\n");
}
h(0);
h(1);
