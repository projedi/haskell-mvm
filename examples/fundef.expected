1 4.34 def
1 4.34 def
0 3.1 abc
2.6
1
